### ISSUE TITLE
Is E226 supposed to add whitespace around all plus signs, even in list slices?

### ISSUE THREAD

**Issue by kaddkaka** [astral-sh/ruff#18715]
### Question

The rules E226 https://docs.astral.sh/ruff/rules/missing-whitespace-around-arithmetic-operator/ seem to make this code change which I think make the code less readable compare to before:

```diff
-time[offset+1:offset+2]
+time[offset + 1:offset + 2]
```

is this intended? Is there any configuration for this rule?

**Comment by ntBre** [comment]
Hmm, I'm not sure about this one. [PEP 8](https://peps.python.org/pep-0008/#pet-peeves) includes at least two different "Correct" formatting examples for related cases:

```python
ham[lower+offset : upper+offset]
ham[lower + offset : upper + offset]
```

`pycodestyle` flags your first case, but running `autopep8` fixes it to the strange-looking

```python
time[offset +1:offset +2]
```

which apparently satisfies `pycodestyle`.

So I think our E226 results are reasonable, but I'd prefer the results from our formatter or from black, which match the second example from PEP 8:

```python
time[offset + 1 : offset + 2]
```

**Comment by MichaReiser** [comment]
Yes, I think we changed this so to make the rule compatible with our formatter.
